src/main/java/thewaypointers/trafficsimulator/gui/DragEventListener.java

```diff
@@ -0,0 +1,75 @@
+package thewaypointers.trafficsimulator.gui;
+
+import javax.swing.*;
+import javax.swing.event.MouseInputListener;
+import java.awt.*;
+import java.awt.event.MouseEvent;
+
+public class DragEventListener implements MouseInputListener {
```

Comment:
I would rename it to PanelMouseDragger, because that's basically what it does - enables dragging of chosen panel by mouse